source/patterns/@aws-solutions-constructs/core/test/elasticsearch-helper.test.ts

```diff
@@ -132,8 +149,154 @@ test('Test override SnapshotOptions for buildElasticSearch', () => {
   });
 });
 
-test('Test override ES version for buildElasticSearch', () => {
+test('Test VPC with 1 AZ, Zone Awareness Disabled', () => {
   const stack = new Stack();
+
+  const vpc = deployVpc(stack);
+
+  deployES(stack, 'test-domain', {
+    elasticsearchClusterConfig: {
+      dedicatedMasterEnabled: true,
+      dedicatedMasterCount: 3,
+      instanceCount: 3,
+      zoneAwarenessEnabled: false
+    }
+  }, undefined, vpc);
+
+  expect(stack).toHaveResourceLike('AWS::Elasticsearch::Domain', {
+    DomainName: "test-domain",
+    ElasticsearchClusterConfig: {
+      DedicatedMasterCount: 3,
+      DedicatedMasterEnabled: true,
+      InstanceCount: 3,
+      ZoneAwarenessEnabled: false
+    }
+  });
+});
+
+test('Test VPC with 2 AZ, Zone Awareness Enabled', () => {
+  const stack = new Stack();
```

Comment:
As there is no code here creating the Elasticsearch Domain, I don't know why you expect 2 AZs. the VPC is deployed with DefaultIsolatedVpcProps, but that doesn't specify 2 AZs. The DeployES() function doesn't seem to enforce 2 AZs. This is the downside to putting all the setup code elsewhere.

I think we do need a test with a VPC that explicitly has 2 AZs and a ClusterConfig to match, but while the result test seems to check this, the setup code is unclear.

Reply:
There is no account defined (environment agnostic), therefore VPC deploys it in 2 AZ by default. I will add an account and put max AZ to 2 to make it clear.